lib/editor/heading-decorator.js

```diff
@@ -0,0 +1,26 @@
+import { SelectionState } from 'draft-js';
+import SimpleDecorator from 'draft-js-simpledecorator';
+
+import Heading from '../components/heading';
+import { headingRegex } from '../note-detail/heading/constants';
+
+const headingDecorator = () => {
+  const strategy = (contentBlock, callback) => {
+    const text = contentBlock.getText();
+    let match;
+
+    while ((match = headingRegex.exec(text)) !== null) {
+      const fullText = text;
+      const headingLevel = match[1].length;
+
+      const start = match.index;
+      const end = start + fullText.length;
+
+      callback(start, end, { headingLevel: headingLevel, content: fullText });
```

Comment:
Did you consider passing the full text as children to the React component? Something about that feels more natural than mimicking it with new prop types…

```js
callback( start, end, { headingLevel, children: fullText } );
```

```js
const Heading = ({ headingLevel, children }) => {
   const elementType = `h${headingLevel}`;
   const elementClass = `heading heading__${headingLevel}`;

   return React.createElement(elementType, { className: elementClass }, children);
 };
```

```js
<Heading headingLevel={level}>Test</Heading>
```

Reply:
These changes have been implemented. Thanks!